addr_manager_ds.go

```diff
@@ -0,0 +1,267 @@
+package peerstore
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	ds "github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+	"github.com/libp2p/go-libp2p-peer"
+	ma "github.com/multiformats/go-multiaddr"
+	mh "github.com/multiformats/go-multihash"
+)
+
+type DatastoreAddrManager struct {
+	ds          ds.Datastore
+	ttlManager  *ttlmanager
+	subsManager *AddrSubManager
+}
+
+func NewDatastoreAddrManager(ctx context.Context, ds ds.Datastore, ttlInterval time.Duration) *DatastoreAddrManager {
+	mgr := &DatastoreAddrManager{
+		ds:          ds,
```

Comment:
Should be wrapped with namespace datastore (https://github.com/ipfs/go-datastore/blob/master/namespace/namespace.go#L39)

Reply:
@magik6k i'm torn between doing that here, or relying on the caller to pass a safe thing for us to use. What convention do you think we should set?

Reply:
ah yeah, my thinking was this instance would be fully unique to the peerstore, as it's relatively cheap to create new badger datastores. but i should 100% have a comment to that effect if that's where we land